synapse/push/pusher.py

```diff
@@ -47,18 +47,18 @@ def __init__(self, hs: "HomeServer"):
 
             logger.info("defined email pusher type")
 
-    def create_pusher(self, pusherdict: Dict[str, Any]) -> Optional[Pusher]:
-        kind = pusherdict["kind"]
+    def create_pusher(self, pusher_config: PusherConfig) -> Optional[Pusher]:
+        kind = pusher_config.kind
         f = self.pusher_types.get(kind, None)
         if not f:
             return None
-        logger.debug("creating %s pusher for %r", kind, pusherdict)
-        return f(self.hs, pusherdict)
+        logger.debug("creating %s pusher for %r", kind, pusher_config)
```

Comment:
does the attrs class give a sane repr?

Reply:
attrs gives decent reprs,  something like (the data is made up...):

```
"PusherConfig(id=1, user_name='@clokep:matrix.org', access_token=2, profile_tag='tag', kind='http', app_id='foobarbaz', app_display_name='Element Web', device_display_name='MBP', pushkey='asdffdsa', ts=1608127939, lang='en', data={'url': 'https://matrix.org/_matrix/push/v1/send'}, last_stream_ordering=0, last_success=1608127939, failing_since=None)"
```